Controllers/ProductController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using EFCoreServices.Models;
+using EFCoreServices.Repository;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace EFCoreServices.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        IProductRepository productRepository;
+        public ProductController(ProductRepository _productRepository)
+        {
+            productRepository = _productRepository;
+        }
+
+        [HttpGet]
+        [Route("GetProductsList")]
+        public async Task<IActionResult> GetProductsList()
+        {
+            try
+            {
+                var products = await productRepository.GetProductsList();
```

Comment:
GetProductsList -> avoid naming like List (type) in methods/variables. 
always use Async to end the name of methods returning type Task ot task<T>